.github/workflows/cs-lint.yml

```diff
@@ -0,0 +1,70 @@
+name: CS & Lint
+
+on:
+  # Run on all pushes and on all pull requests.
+  # Prevent the "push" build from running when there are only irrelevant changes.
+  push:
+    paths-ignore:
+      - "**.md"
+  pull_request:
+  # Allow manually triggering the workflow.
+  workflow_dispatch:
+
+jobs:
+  checkcs:
+    name: "Basic CS and QA checks"
+    runs-on: ubuntu-latest
+
+    env:
+      XMLLINT_INDENT: "	"
+
+    steps:
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: "7.4"
+          coverage: none
+          tools: cs2pr
+
+      # Show PHP lint violations inline in the file diff.
+      # @link https://github.com/marketplace/actions/xmllint-problem-matcher
+      - name: Register PHP lint violations to appear as file diff comments
+        uses: korelstar/phplint-problem-matcher@v1
+
+      # Show XML violations inline in the file diff.
+      # @link https://github.com/marketplace/actions/xmllint-problem-matcher
+      - name: Register XML violations to appear as file diff comments
+        uses: korelstar/xmllint-problem-matcher@v1
+
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      # Validate the composer.json file.
+      # @link https://getcomposer.org/doc/03-cli.md#validate
+      - name: Validate Composer installation
+        run: composer validate --no-check-all
+
+      # Install dependencies and handle caching in one go.
+      # @link https://github.com/marketplace/actions/install-composer-dependencies
+      - name: Install Composer dependencies
+        uses: ramsey/composer-install@v1
+
+      # Lint PHP.
+      - name: Lint PHP against parse errors
+        run: composer lint-ci | cs2pr
+
+      # Needed as runs-on: system doesn't have xml-lint by default.
+      # @link https://github.com/marketplace/actions/xml-lint
+      - name: Lint phpunit.xml.dist
+        uses: ChristophWurst/xmllint-action@v1
+        with:
+          xml-file: ./phpunit.xml.dist
+          xml-schema-file: ./vendor/phpunit/phpunit/phpunit.xsd
+
+      # Check the code-style consistency of the PHP files.
+#      - name: Check PHP code style
+#        continue-on-error: true
+#        run: vendor/bin/phpcs --report-full --report-checkstyle=./phpcs-report.xml
```

Comment:
@trepmal
This is how Edit Flow is implementing to check PHPCS for just diff files. 

https://github.com/Automattic/Edit-Flow/blob/c7e3fa7536b5787d494d54d3e34f240355e06dc6/.github/workflows/php-tests.yml#L56-L60

Ideally, we run PHPCS for all PHP files, but for old plugins like this, it's not productive to detect and fix all PHPCS errors. So I think this approach should be still kept though there may be a better way.